packages/flutter/test/material/app_test.dart

```diff
@@ -833,6 +834,59 @@ void main() {
     expect(find.text('regular page one'), findsNothing);
     expect(find.text('regular page two'), findsNothing);
   });
+
+  testWidgets('MaterialApp propogates DefaultTextStyle properties to children', (WidgetTester tester) async {
+    const String text = 'foo';
+    await tester.pumpWidget(
+      const DefaultTextStyle(
+        style: TextStyle(),
+        textHeightBehavior: TextHeightBehavior(applyHeightToFirstAscent: false),
+        child: MaterialApp(
+          home: Scaffold(
+            body: Text(text),
+          ),
+        ),
+      ),
+    );
+
+    final RenderParagraph renderParagraph = tester.renderObject<RenderParagraph>(find.text(text));
+    expect(renderParagraph.textHeightBehavior.applyHeightToFirstAscent, isFalse);
+  });
+
+  testWidgets('MaterialApp propogates DefaultTextStyle properties to routes', (WidgetTester tester) async {
+    const String pageOneText = 'foo';
+    const String pageTwoText = 'bar';
+    const String pageTwoRoute = '/bar';
+
+    await tester.pumpWidget(
+       DefaultTextStyle(
+        style: const TextStyle(),
+        textHeightBehavior: const TextHeightBehavior(applyHeightToFirstAscent: false),
+        child: MaterialApp(
+          routes: <String, WidgetBuilder>{
+            '/': (BuildContext context) {
+              return Column(children: <Widget>[
```

Comment:
Rather than tapping the button to trigger the route, can you assign the context to a test local variable and use it to get the Navigator and push the route instead of wait tester.tap(find.byType(RaisedButton)).

I think either is OK, but this way it doesn't rely on the button